lib/IDE/Refactoring.cpp

```diff
@@ -2070,6 +2070,211 @@ bool RefactoringActionExpandTernaryExpr::performChange() {
+	isApplicable(ResolvedRangeInfo Info, DiagnosticEngine &Diag) {
+
+  if (Info.Kind != RangeKind::SingleStatement
+      && Info.Kind != RangeKind::MultiStatement)
+    return false;
+
+  if (Info.ContainedNodes.empty())
+    return false;
+
+  IfStmt *If = nullptr;
+
+  if (Info.ContainedNodes.size() == 1) {
+    if (auto S = Info.ContainedNodes[0].dyn_cast<Stmt*>()) {
+        If = dyn_cast<IfStmt>(S);
+    }
+  }
+
+  if (!If)
+    return false;
+
+  auto CondList = If->getCond();
+
+  if (CondList.size() == 1) {
+      auto E = CondList[0];
+      auto P = E.getKind();
+      if (P == swift::StmtConditionElement::CK_PatternBinding)
+        return true;
+  }
+
+  return false;
+}
+	
+bool RefactoringActionConvertIfLetExprToGuardExpr::performChange() {
+
+  IfStmt *If = nullptr;
+
+  if (RangeInfo.ContainedNodes.size() == 1) {
+    if (auto S = RangeInfo.ContainedNodes[0].dyn_cast<Stmt*>()) {
+        If = dyn_cast<IfStmt>(S);
+    }
+  }
+
+  if (!If)
+    return true; // abort
+
+  auto CondList = If->getCond();
+
+  if (CondList.size() != 1)
+    return true; // abort
+
+  auto E = CondList[0];
+  auto P = E.getPatternOrNull();
+  if (!P)
+    return true; // abort
+
+    // Get if-let condition
+    SourceRange range = E.getSourceRange();
+    SourceManager &SM = RangeInfo.RangeContext->getASTContext().SourceMgr;
+    auto CondCharRange = Lexer::getCharSourceRangeFromSourceRange(SM, range);
+
+    auto Body = dyn_cast_or_null<BraceStmt>(If->getThenStmt());
+
+    if (!Body)
+        return true; // abort
+
+    // Get if-let then body.
+    auto firstElement = Body->getElements()[0];
+    auto lastElement = Body->getElements().back();
+    SourceRange bodyRange = firstElement.getSourceRange();
+    bodyRange.widen(lastElement.getSourceRange());
+    auto BodyCharRange = Lexer::getCharSourceRangeFromSourceRange(SM, bodyRange);
+
+    llvm::SmallString<64> DeclBuffer;
+    llvm::raw_svector_ostream OS(DeclBuffer);
+
+    llvm::StringRef Space = " ";
+    llvm::StringRef NewLine = "\n";
+
+    OS << tok::kw_guard << Space;
+    OS << CondCharRange.str().str() << Space;
+    OS << tok::kw_else << Space;
+    OS << tok::l_brace << NewLine;
+
+    // Get if-let else body.
+    auto ElseBody = dyn_cast_or_null<BraceStmt>(If->getElseStmt());
+    if (ElseBody) {
+        auto firstElseElement = ElseBody->getElements()[0];
+        auto lastElseElement = ElseBody->getElements().back();
+        SourceRange elseBodyRange = firstElseElement.getSourceRange();
+        elseBodyRange.widen(lastElseElement.getSourceRange());
+        auto ElseBodyCharRange = Lexer::getCharSourceRangeFromSourceRange(SM, elseBodyRange);
+        OS << ElseBodyCharRange.str().str() << NewLine;
+    }
+
+    OS << tok::kw_return << NewLine;
+    OS << tok::r_brace << NewLine;
+    OS << BodyCharRange.str().str();
+
+    // Replace if-let to guard
+    auto ReplaceRange = RangeInfo.ContentRange;
+    EditConsumer.accept(SM, ReplaceRange, DeclBuffer.str());
+
+	return false;
```

Comment:
Indentation issue here and all other places.

Reply:
Fixed.